datalad/config.py

```diff
@@ -283,16 +294,43 @@ def reload(self, force=False):
                 # overwrite existing value, do not amend to get multi-line
                 # values
                 self._store, self._cfgfiles = _parse_gitconfig_dump(
-                    stdout, self._store, self._cfgfiles, replace=False)
+                    stdout, self._store, self._cfgfiles, replace=False,
+                    cwd=self._runner.cwd)
 
         if self._src_mode == 'dataset':
             # superimpose overrides, and stop early
             self._store.update(self.overrides)
             return
 
-        stdout, stderr = self._run(run_args, log_stderr=True)
-        self._store, self._cfgfiles = _parse_gitconfig_dump(
-            stdout, self._store, self._cfgfiles, replace=True)
+        if self._src_mode == 'dataset-local':
+            run_args.append('--local')
+
+        dspath_exists = self._dataset_path and exists(self._dataset_path)
+        if not dspath_exists:
+            # if the dataset is not here (yet), we prevent reading from
+            # CWD in case this happens to be another dataset
+            run_args.append(
```

Comment:
Shouldn't that happen only if `self._cfgfiles` do indeed include global/system configs?
Since docstring reads "Reload all configuration items from the configured sources". Now, those configured sources aren't actually configured yet when `reload ` is called first by `__init__`. So, may be store `__init__`'s `source` and consider it here?

Reply:
Another thing: `dspath_exists`:
I wonder whether this is sufficient. The question is whether at that location that particular dataset exists, isn't it? Existence of the path doesn't imply that there's some config file to read from or even an actual dataset at that place. Currently not quite sure about the implications, but looking a few lines down, there's `325     cwd=self._dataset_path if dspath_exists else None)` for ` _parse_gitconfig_dump`, while the `_run` call before doesn't get that passed based on the same criteria. Which means, that `_parse_gitconfig_dump` does the file path completion based on the assumption, that the call ran in that CWD, but that's not clear to me.